SEED-platform/seed
# 2451 feat: add analysis pipelines and tasks for preparing bsyncr input

#### Any background context you want to provide?
When an analysis is created, e.g. when hitting the POST /analyses/ endpoint, there is preparation that must be done before the analysis can actually be run.

#### What's this PR do?
- Defines the basic structure of an "analysis pipeline", shared functionality between pipelines
- Creates a pipeline step for preparing the bsyncr analysis by creating BuildingSync files from the selected properties

#### How should this be manually tested?
Run the automated tests. They will generate BuildingSync files at `MEDIA_ROOT/analysis_input_files/<analysis id>/`. You can check to make sure they look ok.

If you _really_ want to test things, you can test it with the current bsyncr server. Setup the server here and follow the setup instructions: https://github.com/BuildingSync/bsyncr-server
Note that building the image might take a while, installing the R devtools is reeeal slow.
Start the server like instructed in the readme
Then post one of the files like so
```bash
curl -F 'file=@<your test file here>' localhost:5000
```
You should get back your original BuildingSync file that's been updated to include model data (see the DerivedModel section). If it failed you'll get back some json describing the error.

#### What are the relevant tickets?
Resolves #2451
